tfjs-converter/src/executor/model_analysis.ts

```diff
@@ -146,26 +145,87 @@ export function getNodesInTopologicalOrder(
   return orderedNodes;
 }
 
-const CONTROL_FLOW_OPS = [
+/**
+ * Given the execution info, return a map from node to the disposable node list
+ * after its execution.
+ *
+ * @returns A map from node to disposable nodes after its
+ *     execution. That is, for a node `x`, `nodeLiveUntilMap[x]` indicates all
+ *     nodes which their intermediate tensors should be disposed after `x` being
+ *     executed.
+ */
+export function getNodeLiveUntilMap(orderedNodes: Node[]): Map<Node, Node[]> {
+  const nNodes = orderedNodes.length;
+  const nodeToOrder = new Map<Node, number>();
+  for (let i = 0; i < nNodes; ++i) {
+    nodeToOrder.set(orderedNodes[i], i);
+  }
+
+  // `liveUntil[i]` indicates that "all the intermediate tensors from
+  // `orderedNodes[i]` should be disposed after `orderedNodes[liveUntil[i]]`
+  // being executed."
+  const INF_LIFE = Math.floor(Number.MAX_SAFE_INTEGER / 2);
```

Comment:
Nit: Why use MAX_SAFE_INTEGER / 2 instead of MAX_SAFE_INTEGER? I don't see any place where INF_LIFE would have something added to it, although I may have missed something.

Reply:
Nice catch. Yes it can just be MAX_SAFE_INTEGER here.

I was testing another algorithm before I sent out this PR and that algorithm can tweak the life by adding some offset to its value. That's why I made it a smaller value to be safe, but it turns out that the one you reviewed is good enough and the life here can just be the max value.